rotkehlchen/tests/exchanges/test_binance.py

```diff
@@ -769,3 +771,41 @@ def mock_response(url, timeout):  # pylint: disable=unused-argument
             )
     assert 'myTrades failed with HTTP status code: 418' in str(e.value)
     assert binance_mock_get.call_args_list == expected_calls
+
+
+def test_binance_query_trade_history_custom_markets(function_scope_binance, user_data_dir):
+    """Test that custom pairs are queried correctly"""
+    msg_aggregator = MessagesAggregator()
+    db = DBHandler(user_data_dir, '123', msg_aggregator, None)
+
+    binance_api_key = ApiKey('binance_api_key')
+    binance_api_secret = ApiSecret(b'binance_api_secret')
+    db.add_exchange('binance', Location.BINANCE, binance_api_key, binance_api_secret)
+
+    binance = function_scope_binance
+
+    markets = ['ETHBTC', 'BNBBTC', 'BTCUSDC']
+    db.set_binance_pairs('binance', markets, Location.BINANCE)
+
+    def counter():
+        count = 0
+        while True:
+            yield count
+            count += 1
+
+    count = counter()
+    p = re.compile(r'symbol=[A-Z]*')
+    seen = set()
```

Comment:
You can do this much simpler by using `nonlocal`. Example here:

https://github.com/rotki/rotki/blob/8256eb4fd72feb12fee09bfc360feb07b6ce8bc0/rotkehlchen/tests/unit/test_inquirer.py#L63-L77

Reply:
Thanks! I was wondering how to improve this test :+1: